conbench/entities/history.py

```diff
@@ -68,15 +72,42 @@ class HistorySerializer:
     many = _Serializer(many=True)
 
 
-def get_history(case_id, context_id, hardware_hash, repo) -> List[tuple]:
+def get_history(
+    case_id: str,
+    context_id: str,
+    hardware_hash: str,
+    repo: str,
+    include_current_commit_in_rolling_stats: bool = True,
+) -> list:
     """Given a case/context/hardware/repo, return all non-errored BenchmarkResults
     (past, present, and future) on the default branch that match those criteria, along
     with information about the stats of the distribution as of each BenchmarkResult.
-    Order is not guaranteed.
+    Order is not guaranteed. Used to power the timeseries plots and z-scores on
+    BenchmarkResults.
+
+    Let a "segment" of the matching BenchmarkResults be defined as all BenchmarkResults
+    between two BenchmarkResults where ``begins_distribution_change`` is True (inclusive
+    on the left, exclusive on the right). The stats of the distribution include:
+
+        - ``rolling_mean`` - the rolling mean of the current segment (calculated using
+          the mean of each BenchmarkResult). This window is an "expanding window." That
+          is, the window will use unlimited past data (as long as it doesn't go outside
+          the segment).
+        - ``residual`` - the difference of the current BenchmarkResult's mean from the
+          rolling mean.
+        - ``rolling_stddev`` - the rolling standard deviation of the residuals. This is
+          also an expanding window, and it always goes back to the beginning of history,
+          even beyond the start of the segment.
```

Comment:
> the rolling standard deviation of the residuals. This is also an expanding window, and it always goes back to the beginning of history, even beyond the start of the segment

Skeptical about this methodological choice (calculating mean and stddev from different sample collections). It's not the right patch to discuss this though. Relates to https://github.com/conbench/conbench/issues/644.

Reply:
Yeah, this is a sugary technique to allow us to start returning z-score information immediately after a marked distribution change. Basically, we're baking in the assumption that while the mean might change after a distribution change, the spread does not change, at least by the magnitude that would make a difference in analysis. This unlocks the ability to provide somewhat-meaningful alerts immediately after recognized regressions or improvements.

With #641 the impact of this baked-in assumption will lessen. It probably will never be perfectly rigorous. But I think the tradeoff is worth it.

Reply:
I added a little to the docstring to explain this choice.